### ISSUE TITLE
Why checkpoint.pth on the output folder is not in compliance with true model size?

### ISSUE THREAD

**Issue by Eurus-Holmes** [facebookresearch/diffq#11]
## ❓ Questions

For example, when I fine-tune pretrained Vit model with LSQ on CIFAR-10 dataset, the output True model size is 41.20 MB, but on the `./outputs` folder, the `checkpoint.th` is 686 MB, why is not in compliance with true model size?

<img width="778" alt="Screen Shot 2022-03-02 at 20 40 18" src="https://user-images.githubusercontent.com/34226570/156497377-876fcb8f-75e5-4a4a-ad14-fde355b2232c.png">

**Comment by adefossez** [comment]
The checkpoint file contains everything required for training. In particular, during training, weights are kept in float32 (this is required as otherwise the small updates from the gradient would never lead to a change of quantized value), as well as all the state of the optimizer (momentum, squared gradients etc). In order to get a shippable small model, you must call at the end of training `solver.quantizer.get_quantized_state()`. If you `torch.save` what is returned, it should have the expected model size.

**Comment by Eurus-Holmes** [comment]
Got it, thanks!
